src/bl_support.h

```diff
@@ -0,0 +1,103 @@
+/*
+ * Scene Connect Bootloader
+ * Copyright (C) 2019 Scene Connect 
+ * https://www.scene.community/
+ *
+ * This code is developed based on the Okra Bootloader from Okra Solar.
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ */
+#ifndef BOOTLOADER_H
+#define BOOTLOADER_H
+
+#pragma once
+
+#include <cstdint>
+
+#include "config.h"
+
+/** Length of the bootloader name string (including \0 termination) */
+const uint8_t BOOTLADER_NAME_LENGTH = 18;
+
+/**
+ * Flash address for the bootloader status structure.
+ * This address must be aligned to a flash page. Each time a new application
+ * is detected by the bootloader (BootloaderState::newApp),
+ * the full page is erased, and the updated struct is written to the address 
+ */
+const uint32_t BOOTLOADER_STATUS_STRUCT_ADDR = 0x0802FF80; // The last flash page
+
+/**
+ * Bootloader state enumeration. This state needs to be set to "newApp"
+ * by the application after an update, and to "stableApp" after the
+ * first successful boot 
+ */
+enum BootloaderState {
+    NO_STATE = 0,     // State not properly initialized
```

Comment:
Remark: If these comments should be considered by Doxygen, you need to use `///< my comment`. Three slashes is the doxygen style comment and the less-than sign signifies that the single-line comment refers to the line before it and not after it.

Reply:
Do you think we need to add this to the guidelines under: 
https://libre.solar/docs/coding_style/

Reply:
Agreed and updated.